marklogic/client.py

```diff
@@ -68,6 +72,62 @@ def prepare_request(self, request, *args, **kwargs):
         request.url = urljoin(self.base_url, request.url)
         return super(Client, self).prepare_request(request, *args, **kwargs)
 
+    def invoke(
+        self, module: str, vars: dict = None, return_response: bool = False, **kwargs
+    ):
+        """
+        Send a script (XQuery or javascript) and possibly a dict of vars
+        to MarkLogic via a POST to the endpoint defined at
+        https://docs.marklogic.com/REST/POST/v1/eval.
+
+        :param module: The URI of a module in the modules database of the app server
+        :param vars: a dict containing variables to include
+        :param return_response: boolean specifying if the entire original response
+        object should be returned (True) or if only the data should be returned (False)
+        upon a success (2xx) response. Note that if the status code of the response is
+        not 2xx, then the entire response is always returned.
+        """
+        if module is None:
```

Comment:
I don't think this is needed - if a Python arg doesn't have a default value, a developer knows that it's required. Without it, in this scenario, the user will get back a 400 with a message of "No module to invoke" which is a good error message. 

If a user invokes `client.invoke()`, they'll get a nice error from Python. 

So I think it's better not to do this - particularly as this isn't being done in `documents.py`, and I don't want to establish a convention that non-optional args must have "if None" checks.